### DISCUSSION TITLE
How to disable email confirm in Self-Hosted

### DISCUSSION THREAD

**Discussion by jgreywolf** [supabase/supabase#18394] [Questions]
I think the title is pretty self explanatory.  But I will add more...

I have seen various discussions such as https://github.com/orgs/supabase/discussions/4444, which explains that the settings UI is not going to be available in the self-hosted version of supabase, instead making use of config files.

So - how do we configure the app to not require email confirmation before allowing users to sign in?

I have customers that are using a self-hosted version of our app. Due to security restrictions they cannot/will not add email to supabase, so we want to turn off email verification in these scenarios.

In which case, how do we turn off email confirmation for self hosted uses?

As an aside, `ENABLE_EMAIL_AUTOCONFIRM=true` does not seem to be working. And even more annoying is that the kubernetes/helm chart version doesnt seem to even expose this setting...

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/self-hosting/auth/config

Also note that an email is still required even if confirm is off.  It just can be a valid email format fake one.

**Reply by jgreywolf** [reply]
`auth.email.enable_confirmations` is currently set to false, but the site is still requiring email addresses to be confirmed before you can login.

Also, is this accessible in the containerized version?

**Reply by jgreywolf** [reply]
I wish I could create a user in the client similar to
```
 const { data, error } = await supabase.auth.admin.createUser({
  email: 'user@email.com',
  password: password,
  email_confirm: true
})
```

**Reply by GaryAustin1** [reply]
Sorry, I'm at the limit of my knowledge on this as I don't do self-hosting. 
Your last comment would not be secure if run on the browser as the user could change it.

**Reply by jgreywolf** [reply]
Is it possible to update the auth record via code to set as confirmed?

**Reply by GaryAustin1** [reply]
I believe if you set the email_confirmed_at column in auth.users that will confirm the user.  Note that is the way auth currently seems to work. You could check the source code in supabase/gotrue, but it is not a documented approach so could change in the future.

**Reply by jgreywolf** [reply]
Can I use the following to update a record in the auth schema?

```
const { data, error } = await supabase
  .from('countries')
  .update({ name: 'Australia' })
  .eq('id', 1)
  .select()
```

**Reply by GaryAustin1** [reply]
Not sure what you are asking.  You can't add tables to the auth schema and you can't access the auth schema from the API.
But the code you show seems to be accessing a table in the public schema called countries and does not involve the auth schema at all.

**Reply by jgreywolf** [reply]
Im sorry, I pasted an example of using the update function from the supabase documentation

What I am asking is if I can make use of the "update" function mentioned there to update a record in the auth.users table:

```
 const userUpdate = await data.supabase
        .from('auth.users')
        .update({ email_confirmed_at: DateTime.now() })
        .eq('id', 1)
        .select();
```

(but yes, it tries to access public.auth.users, so not the right thing)

**Reply by GaryAustin1** [reply]
No you can't.  No access to auth directly from API.  You can use an rpc call though to a security definer postgres function to do it.  

But you need to make sure the function is limited to service_role calling it.  https://supabase.com/docs/guides/database/functions#function-privileges

You actually need to revoke public,anon and authenticated roles from the function.  Then only service role can access it.

Edge functions are another possibility.

**Reply by jgreywolf** [reply]
Thanks for the help so far.  It seems like it shouldnt be so hard to turn this feature off for the containerized version

**Reply by gbusto** [reply]
I had this issue of trying to disable email confirmation on self-hosted. Even with GOTRUE_MAILER_AUTOCONFIRM set to false, it would still try to send one and give me this error: `Signup error: Error sending confirmation email`. I had to tweak the .env file email auth section to look like this:
```bash
## Email auth
ENABLE_EMAIL_SIGNUP=true
ENABLE_EMAIL_AUTOCONFIRM=false
#SMTP_ADMIN_EMAIL=admin@example.com
#SMTP_HOST=supabase-mail
SMTP_PORT=2500
#SMTP_USER=fake_mail_user
#SMTP_PASS=fake_mail_password
#SMTP_SENDER_NAME=fake_sender
ENABLE_ANONYMOUS_USERS=false
```

Now I don't get that error and things seem to work. But note that SMTP_PORT still needs to be uncommented. Having an empty value for the port caused the supabase-auth service to fail.

It's been a bit of minefield trying to get things setup in self-hosted Supabase. Keeping a list of everything extra I'm having to do so I can share it somewhere to save people some sanity.

**Reply by Bvcmbru2024** [reply]
I'm about to lose mine, I made a beautiful piece on bolt.new and now cant create a user anymore

**Reply by gbusto** [reply]
EDIT - I was getting warnings, so the easier thing to do is just set them to blank strings:
```bash
## Email auth
ENABLE_EMAIL_SIGNUP=true
ENABLE_EMAIL_AUTOCONFIRM=true
SMTP_ADMIN_EMAIL=
SMTP_HOST=
SMTP_PORT=2500
SMTP_USER=
SMTP_PASS=
SMTP_SENDER_NAME=
ENABLE_ANONYMOUS_USERS=false
```

Also, set `ENABLE_EMAIL_AUTOCONFIRM` to `false` so new users are auto-confirmed.

@Bvcmbru2024 , sorry to hear that. I'm still running into issues too with other stuff. I just spent ~1 hour debugging another weird issue that hadn't come up before. Something to do with SSL and trying to generate types using the supabase cli.

I just shared a quick blog post with my notes and stuff. See if any of this helps you:
https://blog.gbusto.com/self-hosting-supabase-part-1/

**Reply by uplg** [reply]
Tested right now on a self-hosted Supabase, and SMTP params needs to be there for ENABLE_EMAIL_AUTOCONFIRM=true to work.
